libutils/file_lib.c

```diff
@@ -67,6 +68,112 @@ Writer *FileRead(const char *filename, size_t max_size, bool *truncated)
     return w;
 }
 
+bool File_Copy(const char *src, const char *dst)
+{
+    assert(src != NULL);
+    assert(dst != NULL);
+
+    Log(LOG_LEVEL_INFO, "Copying: '%s' -> '%s'", src, dst);
+
+    FILE *in = fopen(src, "r");
+    if (in == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Could not open '%s' (%s)", src, strerror(errno));
+        return false;
+    }
+
+    FILE *out = fopen(dst, "w");
+    if (out == NULL)
+    {
+        Log(LOG_LEVEL_ERR, "Could not open '%s' (%s)", dst, strerror(errno));
+        fclose(in);
+        return false;
+    }
+
+    size_t bytes_in = 0;
+    size_t bytes_out = 0;
+    bool ret = true;
+    do
+    {
+#define BUFSIZE 1024
+        char buf[BUFSIZE] = {0};
+
```

Comment:
Not that performance matters too much but is there a reason to initialize the buf variable inside the block instead of outside of it?

Reply:
Nothing too strong. We tend to prefer to have the variables be as local as possible, for readability / maintainability. It is also safer to make sure it's never dirty. Doing it outside the loop for performance would be valid as well, but I don't think performance is that critical here. IO performance (fwrite and fread) probably makes this difference completely negligible.